src/main/java/randoop/util/Randomness.java

```diff
@@ -171,18 +172,22 @@ public static int nextRandomInt(int i) {
    *
    * @param list the list of elements to select from
    * @param weights the map of elements to their weights; uses the intrinsic weight if the element
-   *     is not a key in the map
+   *     is not a key in the map. Each element's weight must be non-negative. An element with a
+   *     weight of zero will never be selected.
    * @param <T> the type of the elements in the list
    * @return a randomly selected element from {@code list}
    */
   public static <T extends WeightedElement> T randomMemberWeighted(
       SimpleList<T> list, Map<T, Double> weights) {
 
+    // Subset of elements from the input list whose weights are nonzero. The selected element
+    // will be chosen from this list.
+    List<T> listOfElementsWithNonZeroWeights = new ArrayList<>();
```

Comment:
This code was already very inefficient.  It's crazy to perform an O(n) traversal, creating a whole new array just for the purpose of running an O(log n) binary search on it.

Suppose there are 10,000 methods under test.  Then the old code created a 10,000-element array every time it wanted to select a method.  The new code is slightly worse, because an ArrayList is more expensive than an array.

Neither an array nor an ArrayList is necessary.
It's enough to
 * determine the total weight -- this can be maintained by the client and passed in.
 * choose a value that is no more than that, and
 * walk through the elements one-by-one, maintaining a running weight and return the element that makes the total weight greater than the chosen value.

The big benefit is that there is no need to allocate an array or ArrayList.  A smaller benefit is that on average, only half of the elements need to be touched before one can be returned.

I will create another pull request to change Randomness.
(These changes to Randomness need to be undone, in order to avoid conflicts.)

The second part is to change Bloodhound to cache and update the total weight, which will improve efficiency.